client-v2/src/actions/ArticleFragments.ts

```diff
@@ -124,19 +125,29 @@ const maybeInsertGroupArticleFragment = (
           You can proceed, and the last article in the collection will be
           removed automatically, or you can cancel and remove articles from the
           collection yourself.`,
-          // if the user accepts then remove the moved item (if there was one),
+          // if the user accepts, then remove the moved item (if there was one),
           // remove article fragments past the cap count and finally persist
-          (removeAction ? [removeAction] : []).concat([
-            insertGroupArticleFragment(id, index, articleFragmentId),
-            maybeAddFrontPublicationDate(articleFragmentId),
-            addPersistMetaToAction(capGroupSiblings, {
-              id: articleFragmentId,
-              persistTo,
-              applyBeforeReducer: true
-            })(id, collectionCap)
-          ]),
+          () => {
+            const actions = [];
+
+            if (removeAction) {
+              actions.push(removeAction);
+            }
+
+            actions
+              .concat([
+                insertGroupArticleFragment(id, index, articleFragmentId),
+                maybeAddFrontPublicationDate(articleFragmentId),
+                addPersistMetaToAction(capGroupSiblings, {
+                  id: articleFragmentId,
+                  persistTo,
+                  applyBeforeReducer: true
+                })(id, collectionCap)
+              ])
+              .forEach(action => dispatch(action));
```

Comment:
why are we dispatching now?

Reply:
We're moving away from sending arrays of actions for the modal to dispatch on completion to using just a call back.

The downside of this is time travelling through redux wont work very well but following a conversation with Jon we intend to move over to thread's way of handling modals.